app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestActivity.java

```diff
@@ -0,0 +1,177 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.cmput301f20t21.bookfriends.R;
+import com.cmput301f20t21.bookfriends.entities.Book;
+import com.cmput301f20t21.bookfriends.entities.Request;
+import com.cmput301f20t21.bookfriends.ui.library.OwnedListFragment;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RequestActivity extends AppCompatActivity implements ConfirmDialog.ConfirmDialogListener {
+    private RecyclerView recyclerView;
+    private RequestAdapter requestAdapter;
+    private final ArrayList<Request> requestDataList = new ArrayList<>();
+    private RecyclerView.LayoutManager layoutManager;
+    private TextView titleTextView;
+    private TextView authorTextView;
+    private TextView descriptionTextView;
+    private TextView bookStatus;
+    private ImageView bookImage;
+
+    private RequestViewModel requestViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.view_all_requests_activity);
+
+        // getting book ID from previous activity
+        String bookId = getIntent().getStringExtra(OwnedListFragment.VIEW_REQUEST_KEY);
+        displayBookInfo(bookId);
+        displayRequest(bookId);
+        // set up the back button
+        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_baseline_arrow_back_ios_white_18);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        buildRecyclerView();
+    }
+
+    /**
+     * display book information by getting it from FireStore based on bookId
+     * @param bookId
+     */
+    public void displayBookInfo(String bookId) {
```

Comment:
I don't think we should abstract these functions under displayBookInfo(). Having them in onCreate() is fine since they're only called once. If we really want to keep onCreate tidy, a helper function `bindView` could be added to help bind all the views.

Reply:
a better practice is that we deliberately not wrap around any `.observe(this, item -> {...})` because they represents a data binding and should be the top priority to get noticed, thus inside onCreate is best. wrapping them elsewhere without a distinguishable name might cause confusion.